### PR TITLE
Update Kconfig and RTL8730E defconfig files

### PR DESCRIPTION
**Pull request by amogh-samsung**
Update RTL8730E defconfig files
- Update default configurations for RTL8730E
- Enable/disable configurations as per latest use and requirement.

Update Kconfig files with regards to RTL8730E
- Update Kconfig files for making CONFIG selections for RTL8730E.

circleci: Update test configuration for RTL8730E
- Names of defconfigs hello_8M and loadable_apps_8M are updated to hello
 and loadable_apps, respecitvely.
- Modify circleci test script accordingly to use these new defconfigs.

build/configs/rtl8730e: Enable neccessary configs for jSmartFS
- Enable CONFIG_MTD_BYTE_WRITE which helps in journal update efficiency
- Enable 16-bit CRC and Journaling for jSmartFS

build/configs/rtl8730e: Update specific RTL8730E defconfigs as required
- Disable CONFIG_DISABLE_OS_API
- Update CONFIG_MAX_WDOGPARMS=2 -> 4, CONFIG_PREALLOC_WDOGS=8 -> 32
 CONFIG_WDOG_INTRESERVE=0 -> 4 and CONFIG_PREALLOC_TIMERS=2 -> 8
- Enable CONFIG_LOG_DUMP
- Update CONFIG_SCHED_LPWORKSTACKSIZE=2048 -> 4096
- Disable CONFIG_SYSTEM_IPERF
- Disable CONFIG_ENABLE_KILL
- Disable CONFIG_ENABLE_KILLALL
- Disable CONFIG_SCHED_CPULOAD and CONFIG_ENABLE_CPULOAD

build/configs/rtl8730e: Align defconfigs with generated menuconfig outputs
- Remove unnecessary options from defconfigs which are not present when generated using menuconfig option

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@amogh-samsung Let's apply Kconfig change first. Kconfig is different so that this is still incorrect. 
And there is not mt chipset here. It means you don't use menuconfig here.

**Comment by amogh-samsung** [comment]
> @amogh-samsung Let's apply Kconfig change first. Kconfig is different so that this is still incorrect. And there is not mt chipset here. It means you don't use menuconfig here.

I added the Kconfig changes also now.
All defconfigs other than loadable do not have MT7686 config because they are not present in Ecode. The aim with the first commit is to sync defconfig and Kconfig changes from Ecode only.

At the same time, I agree with your idea that MT7686 config should be present in defconfig as per menuconfig selection. Let me work on that too, but in a separate commit please.

**Comment by amogh-samsung** [comment]
NOTE: The first commit also changes the names of hello_8M -> hello and loadable_apps_8M -> loadable_apps to keep them in accordance with Ecode.
Please give your opinion if you feel otherwise

**Comment by sunghan-chang** [comment]
@amogh-samsung Here is public, no one knows the ecode. Let's remove that word from the commit.

**Comment by sunghan-chang** [comment]
@amogh-samsung Sorry I don't get your point. But we can't add MT7686 in the public so that defconfig should not have it.

> At the same time, I agree with your idea that MT7686 config should be present in defconfig as per menuconfig selection. Let me work on that too, but in a separate commit please.

**Comment by sunghan-chang** [comment]
@amogh-samsung You should modify ci setting together.

>NOTE: The first commit also changes the names of hello_8M -> hello and loadable_apps_8M -> loadable_apps to keep them in accordance with Ecode. (This does cause CircleCi test failures because the 2 configs are removed).
Please give your opinion if you feel otherwise

**Comment by amogh-samsung** [comment]
> @amogh-samsung Here is public, no one knows the ecode. Let's remove that word from the commit.

Understood and done

**Comment by amogh-samsung** [comment]
> @amogh-samsung Sorry I don't get your point. But we can't add MT7686 in the public so that defconfig should not have it.
> 
> > At the same time, I agree with your idea that MT7686 config should be present in defconfig as per menuconfig selection. Let me work on that too, but in a separate commit please.

Understood and Done

**Comment by amogh-samsung** [comment]
> @amogh-samsung Let's split the commit for Kconfig change and defconfig change.

Done

**Comment by amogh-samsung** [comment]
> @amogh-samsung You should modify ci setting together.
> 
> > NOTE: The first commit also changes the names of hello_8M -> hello and loadable_apps_8M -> loadable_apps to keep them in accordance with Ecode. (This does cause CircleCi test failures because the 2 configs are removed).
> > Please give your opinion if you feel otherwise

Done

File: os/arch/arm/Kconfig [hunk 1]
```diff
@@ -266,6 +267,8 @@ config ARCH_CORTEXA32
         default n
         select ARCH_DCACHE
         select ARCH_ICACHE
+	select ARCH_HAVE_IRQPRIO
+	select ARCH_HAVE_RESET
```
Comment (jylee9613) [inline os/arch/arm/Kconfig:271]: It's minor, please align the indentation.

**Review by sunghan-chang — COMMENTED** [review]
Let's remove `Reverse Sync - #/3` from the commit. It does not give any meaning here.
And please leave why those changes are necessary in the commit description.

File: os/arch/arm/Kconfig (hunk 1 above)
Comment (amogh-samsung) [reply os/arch/arm/Kconfig:271]: Good catch, thankyou

**Comment by amogh-samsung** [comment]
> Let's remove `Reverse Sync - #/3` from the commit. It does not give any meaning here. And please leave why those changes are necessary in the commit description.

Updated

File: build/configs/rtl8730e/hello/defconfig [hunk 2]
```diff
@@ -24,6 +24,7 @@ CONFIG_BUILD_FLAT=y
 # CONFIG_APP_BINARY_SEPARATION is not set
```
Comment (kishore-sn) [inline build/configs/rtl8730e/hello/defconfig:24]: Why are we renaming loadable and hello 8M configs only?
I see that there are 8M configs for others such as cxxtest, etc.. Better to have a standard convention.

File: build/configs/rtl8730e/loadable_apps_16M/defconfig [hunk 3]
```diff
@@ -290,7 +294,7 @@ CONFIG_ARCH_BOARD_HAVE_FLASH=y
 #
 CONFIG_FLASH_PARTITION=y
 CONFIG_FLASH_MINOR=0
-CONFIG_FLASH_PART_SIZE="48,52,12,400,1844,4808,384,1844,4808,384,512,1280,8,"
+CONFIG_FLASH_PART_SIZE="48,52,12,400,1844,4936,384,1844,4936,384,512,1024,8,"
```
Comment (ewoodev) [inline build/configs/rtl8730e/loadable_apps_16M/defconfig:297]: Please check recently partition size

https://github.com/Samsung/TizenRT/pull/5964/commits/373135ac024bb4d4aef03ef4fc65e0db30e1226d

Comment (ewoodev) [reply build/configs/rtl8730e/loadable_apps_16M/defconfig:297]: I think, This sync is duplicating, please check what is correct...

File: build/configs/rtl8730e/hello/defconfig (hunk 2 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/hello/defconfig:24]: I wish to rename those two files at least to make them aligned with our private files.
The others are added here for specific testing purposes and may be renamed.
Would like to know the general opinion on the naming of the other files @sunghan-chang @kishore-sn @ewoodev

File: build/configs/rtl8730e/loadable_apps_16M/defconfig (hunk 3 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps_16M/defconfig:297]: I understand your concern. There could be some confusion about the latest update partition sizes for use.
I'll check and update

File: build/configs/rtl8730e/loadable_apps_16M/defconfig [hunk 4]
```diff
@@ -315,7 +320,7 @@ CONFIG_HW_ECDSA_VERIFICATION=y
 # CONFIG_HW_RSA_ENC is not set
 CONFIG_HW_SE_STORAGE=y
 # CONFIG_HW_AES_ENC is not set
-CONFIG_SE_SECURE_CONTEXT_SIZE=9216
+CONFIG_SE_SECURE_CONTEXT_SIZE=6144
```
Comment (ewoodev) [inline build/configs/rtl8730e/loadable_apps_16M/defconfig:323]: Please confirm this configuration value.

File: build/configs/rtl8730e/loadable_apps_16M/defconfig [hunk 5]
```diff
@@ -420,8 +425,8 @@ CONFIG_SIG_SIGWORK=17
 #
 # POSIX Message Queue Options
 #
-CONFIG_PREALLOC_MQ_MSGS=4
-CONFIG_MQ_MAXMSGSIZE=600
+CONFIG_PREALLOC_MQ_MSGS=32
```
Comment (ewoodev) [inline build/configs/rtl8730e/loadable_apps_16M/defconfig:428]: Please confirm this configuration value.

Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps_16M/defconfig:428]: These CONFIGs have been recently updated in private with a synchronization PR.
I will check all updates and make corresponding changes here.
Thankyou

**Comment by sunghan-chang** [comment]
@amogh-samsung 
```
[Update Kconfig files with regards to RTL8730E](https://github.com/Samsung/TizenRT/pull/5964/commits/95fd6c81817efa6304039291abd0fa9c70b634a6) 
[95fd6c8](https://github.com/Samsung/TizenRT/pull/5964/commits/95fd6c81817efa6304039291abd0fa9c70b634a6)
- Update Kconfig files for including selections based on RTL8730E.
- This commit squashes and includes changes from the following commits:
   3e4739662e19ae4b3899a35990aeb5e223c3a679
   616df3384d7239f6bb7c9f9951f49eb481eab6b2
```
As I said before, noone knows those commits in public. Could you leave what and why you update them in description?
Same as other commits.

File: build/configs/rtl8730e/cxxtest/defconfig [hunk 6]
```diff
@@ -311,7 +313,7 @@ CONFIG_HW_RSA_ENC=y
 CONFIG_HW_SE_STORAGE=y
 # CONFIG_HW_AES_ENC is not set
 CONFIG_SE_SECURE_CONTEXT_SIZE=9216
-CONFIG_SECURITY_LEVEL=y
+# CONFIG_SECURITY_LEVEL is not set
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/cxxtest/defconfig:316]: Let's keep original.

File: build/configs/rtl8730e/cxxtest/defconfig [hunk 7]
```diff
@@ -1377,6 +1392,7 @@ CONFIG_SYSTEM_NETDB_STACKSIZE=2048
 CONFIG_SYSTEM_NETDB_PRIORITY=100
 # CONFIG_SYSTEM_RAMTEST is not set
 # CONFIG_SYSTEM_READLINE is not set
+# CONFIG_READLINE_ECHO is not set
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/cxxtest/defconfig:1395]: I think you don't modify this using menuconfig. Because when `CONFIG_SYSTEM_READLINE` is disabled, this line should not be added.

File: build/configs/rtl8730e/tc/defconfig [hunk 8]
```diff
@@ -1317,11 +1317,6 @@ CONFIG_EXAMPLES_SECURITY_API_TEST=y
 # CONFIG_EXAMPLES_SMARTFS_POWERCUT is not set
 # CONFIG_EXAMPLES_ST_THINGS is not set
 CONFIG_EXAMPLES_TESTCASE=y
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/tc/defconfig:1319]: Regardless of this PR,
this defconfig is `tc`, but no sub-tc is enabled.

Comment (amogh-samsung) [reply build/configs/rtl8730e/tc/defconfig:1319]: I understand you point :P
What do you suggest please?

File: build/configs/rtl8730e/cxxtest/defconfig (hunk 6 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/cxxtest/defconfig:316]: Done

File: build/configs/rtl8730e/cxxtest/defconfig (hunk 7 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/cxxtest/defconfig]: Sorry for this, I have tried to align the defconfigs with menuconfig.
If anything else is incorrect, please let me know and I will fix it.

File: build/configs/rtl8730e/tc/defconfig (hunk 8 above)
Comment (sunghan-chang) [reply build/configs/rtl8730e/tc/defconfig:1319]: Let's enable kernel tc.

File: build/configs/rtl8730e/hello/defconfig [hunk 9]
```diff
@@ -309,7 +310,7 @@ CONFIG_HW_RSA_ENC=y
 CONFIG_HW_SE_STORAGE=y
 # CONFIG_HW_AES_ENC is not set
 CONFIG_SE_SECURE_CONTEXT_SIZE=9216
-CONFIG_SECURITY_LEVEL=y
+# CONFIG_SECURITY_LEVEL is not set
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/hello/defconfig:313]: Let's revert this change in all defconfigs you modified.

File: build/configs/rtl8730e/loadable_apps/defconfig [hunk 10]
```diff
@@ -462,13 +468,19 @@ CONFIG_GPIO=y
...
 # CONFIG_RTC_IOCTL is not set
-# CONFIG_WATCHDOG is not set
+
+CONFIG_WATCHDOG=y
+CONFIG_WATCHDOG_DEVPATH="/dev/watchdog0"
+# CONFIG_WATCHDOG_FOR_IRQ is not set
 # CONFIG_TIMER is not set
 CONFIG_MMINFO=y
 CONFIG_PRODCONFIG=y
+CONFIG_KLOGBUFFER=y
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/loadable_apps/defconfig:483]: Please update this file using menuconfig. Some configs should be not added.

File: build/configs/rtl8730e/hello/defconfig (hunk 9 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/hello/defconfig:313]: Done

File: build/configs/rtl8730e/tc/defconfig (hunk 8 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/tc/defconfig:1319]: Done

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 10 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig]: Updated all defconfigs based on menuconfig.
Have tried to align the selections, please take a look.

File: build/configs/rtl8730e/loadable_apps/defconfig [hunk 11]
```diff
@@ -165,6 +167,8 @@ CONFIG_AMEBASMART_WIFI=y
 # Realtek RTL8730E BLE Support
 #
 CONFIG_AMEBASMART_BLE=y
+# CONFIG_FTL_ENABLED is not set
+CONFIG_BLE_INDICATION=y
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/loadable_apps/defconfig:171]: I just search `CONFIG_BLE_INDICATION` and find it's amebad's config, not amebasmart.
I think this adding is wrong.

File: build/configs/rtl8730e/loadable_apps/defconfig [hunk 12]
```diff
@@ -181,8 +185,8 @@ CONFIG_AMEBASMART_TRUSTZONE=y
 # CONFIG_ARCH_NOINTC is not set
 # CONFIG_ARCH_VECNOTIRQ is not set
 # CONFIG_ARCH_DMA is not set
-# CONFIG_ARCH_HAVE_IRQPRIO is not set
-CONFIG_ARCH_L2CACHE=y
+CONFIG_ARCH_HAVE_IRQPRIO=y
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/loadable_apps/defconfig:188]: You don't enable this in the flat. I think this should be the same regardless of flat or loadable.

File: build/configs/rtl8730e/loadable_apps/defconfig [hunk 13]
```diff
@@ -193,11 +197,12 @@ CONFIG_ARCH_HAVE_MMU=y
 # CONFIG_ARCH_NAND_HWECC is not set
 # CONFIG_ARCH_HAVE_EXTCLK is not set
 # CONFIG_ARCH_HAVE_POWEROFF is not set
-# CONFIG_ARCH_HAVE_RESET is not set
+CONFIG_ARCH_HAVE_RESET=y
 CONFIG_ARCH_USE_MMU=y
 CONFIG_ARCH_HAVE_FETCHADD=y
 # CONFIG_ARCH_ADDRENV is not set
 # CONFIG_PAGING is not set
+CONFIG_ARCH_IRQPRIO=y
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/loadable_apps/defconfig:205]: same as previous

File: build/configs/rtl8730e/loadable_apps/defconfig [hunk 14]
```diff
@@ -849,15 +851,14 @@ CONFIG_LWIP_DHCPS_MAX_STATION_NUM=8
...
 # HW Options
 #
-# CONFIG_TLS_HW_RNG is not set
-# CONFIG_TLS_HW_DH_PARAM is not set
-# CONFIG_TLS_HW_ECDH_PARAM is not set
-# CONFIG_TLS_HW_ECDSA_VERIFICATION is not set
+CONFIG_TLS_HW_RNG=y
+CONFIG_TLS_HW_DH_PARAM=y
+CONFIG_TLS_HW_ECDH_PARAM=y
+CONFIG_TLS_HW_ECDSA_VERIFICATION=y
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/loadable_apps/defconfig:861]: @jylee9613 Could you confirm this is right or not?

File: build/configs/rtl8730e/loadable_apps/defconfig [hunk 15]
```diff
@@ -1158,7 +1167,11 @@ CONFIG_LIBC_STRERROR=y
 CONFIG_LIBC_TMPDIR="/tmp"
 CONFIG_LIBC_MAX_TMPFILE=32
 CONFIG_ARCH_LOWPUTC=y
-# CONFIG_LIBC_LOCALTIME is not set
+CONFIG_LIBC_LOCALTIME=y
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/loadable_apps/defconfig:1170]: As far as I know, we disable this.
@ewoodev Could you confirm this?

File: build/configs/rtl8730e/loadable_apps/defconfig [hunk 16]
```diff
@@ -1228,13 +1258,11 @@ CONFIG_BINFMT_ENABLE=y
...
 CONFIG_ELF_ALIGN_LOG2=2
-CONFIG_ELF_STACKSIZE=2048
+CONFIG_ELF_STACKSIZE=8192
 CONFIG_ELF_BUFFERSIZE=32
 CONFIG_ELF_BUFFERINCR=32
 CONFIG_ELF_EXCLUDE_SYMBOLS=y
-CONFIG_ELF_CACHE_READ=y
-CONFIG_ELF_CACHE_BLOCK_SIZE=2048
-CONFIG_ELF_CACHE_BLOCKS_COUNT=60
+# CONFIG_ELF_CACHE_READ is not set
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/loadable_apps/defconfig:1265]: Do you want to disable this? As I remember, your team enabled.
Could you confirm this with your team?

File: build/configs/rtl8730e/loadable_apps/defconfig [hunk 17]
```diff
@@ -1439,10 +1404,12 @@ CONFIG_EXAMPLES_WIFIMANAGER_AP_LIST_ITEMS_COUNT=10
 CONFIG_TASH=y
 CONFIG_TASH_MAX_STORE_COMMANDS=10
 # CONFIG_TASH_USLEEP is not set
-# CONFIG_TASH_COMMAND_INTERFACE is not set
-CONFIG_TASH_CMDTASK_STACKSIZE=4096
+CONFIG_TASH_REBOOT=y
+CONFIG_TASH_COMMAND_INTERFACE=y
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/loadable_apps/defconfig:1408]: Let's keep disabled.

**Review by sunghan-chang — COMMENTED** [review]
Could you check other loadable defconfigs as well?

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 14 above)
Comment (jylee9613) [reply build/configs/rtl8730e/loadable_apps/defconfig:861]: These configs are related to mbedtls library, and they were enabled in commercial rtl8721csm. I think it's better to enable these configs.

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 15 above)
Comment (ewoodev) [reply build/configs/rtl8730e/loadable_apps/defconfig:1170]: This need to be enabled.

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 11 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig:171]: Understood, removed

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 12 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig:188]: Enabled in all defconfigs, added default selection for Cortex A32 as well.

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 13 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig:205]: Enabled in all defconfigs, added default selection for Cortex A32 as well.

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 14 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig:861]: @jylee9613 these should be enabled in all defconfigs right?

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 15 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig:1170]: @ewoodev this should be enabled in all defconfigs then?

Comment (ewoodev) [reply build/configs/rtl8730e/loadable_apps/defconfig:1170]: yes

```
CONFIG_LIBC_LOCALTIME=y
CONFIG_LIBC_TZ_MAX_TIMES=320
CONFIG_LIBC_TZ_MAX_TYPES=20
CONFIG_LIBC_TZDIR="/etc/zoneinfo"
# CONFIG_LIBC_DOWNLOAD_ZONEINFO is not set
```

Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig:1170]: Updated all

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 17 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig:1408]: Done

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 16 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig]: Confirmed, should be enabled

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 14 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig:860]: @jylee9613 , requesting confirmation on this please.

**Comment by amogh-samsung** [comment]
Powercut test completed with JOURNALING and BYTE_WRITE enabled

> Started: 15 November 2023
> Status as of: 20 November 2023,
> No. of iterations: 15,000
> Total Read/Write size: ~200KiB x 15,000 itr = ~3,000,000 KiB (3.0 million KiB approximately, each)
> Failures: None

File: build/configs/rtl8730e/loadable_apps/defconfig (hunk 14 above)
Comment (jylee9613) [reply build/configs/rtl8730e/loadable_apps/defconfig:861]: Sorry for late apply. It should be enabled for onboarding.

Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps/defconfig:861]: Done

File: build/configs/rtl8730e/cxxtest/defconfig [hunk 18]
```diff
@@ -958,7 +958,7 @@ CONFIG_MTD_FTL=y
 # MTD_FTL Configurations
 #
 # CONFIG_MTD_CONFIG is not set
-# CONFIG_MTD_BYTE_WRITE is not set
+# CONFIG_MTD_BYTE_WRITE=y
```
Comment (jylee9613) [inline build/configs/rtl8730e/cxxtest/defconfig:959]: It's comment out. Please remove '#'

File: build/configs/rtl8730e/hello/defconfig [hunk 19]
```diff
@@ -957,7 +957,7 @@ CONFIG_MTD_FTL=y
 # MTD_FTL Configurations
 #
 # CONFIG_MTD_CONFIG is not set
-# CONFIG_MTD_BYTE_WRITE is not set
+# CONFIG_MTD_BYTE_WRITE=y
```
Comment (jylee9613) [inline build/configs/rtl8730e/hello/defconfig:958]: Same as above.

File: build/configs/rtl8730e/hello_16M/defconfig [hunk 20]
```diff
@@ -957,7 +957,7 @@ CONFIG_MTD_FTL=y
 # MTD_FTL Configurations
 #
 # CONFIG_MTD_CONFIG is not set
-# CONFIG_MTD_BYTE_WRITE is not set
+# CONFIG_MTD_BYTE_WRITE=y
```
Comment (jylee9613) [inline build/configs/rtl8730e/hello_16M/defconfig:958]: Same as above.

File: build/configs/rtl8730e/tc/defconfig [hunk 21]
```diff
@@ -957,7 +957,7 @@ CONFIG_MTD_FTL=y
 # MTD_FTL Configurations
 #
 # CONFIG_MTD_CONFIG is not set
-# CONFIG_MTD_BYTE_WRITE is not set
+# CONFIG_MTD_BYTE_WRITE=y
 
 #
```
Comment (jylee9613) [inline build/configs/rtl8730e/tc/defconfig:959]: Same as above.

File: build/configs/rtl8730e/cxxtest/defconfig (hunk 18 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/cxxtest/defconfig]: Done, in all files

File: build/configs/rtl8730e/cpp_loadable/defconfig [hunk 22]
```diff
@@ -1234,7 +1271,7 @@ CONFIG_ELF_CACHE_BLOCKS_COUNT=60
 CONFIG_BINFMT_CONSTRUCTORS=y
```
Comment (jylee9613) [inline build/configs/rtl8730e/cpp_loadable/defconfig:1271]: If you want to enable this, you should enable SCHED_STARTHOOK. BINFMT_CONSTRUCTORS depends on SCHED_STARTHOOK.

Comment (jylee9613) [reply build/configs/rtl8730e/cpp_loadable/defconfig:1271]: Also, please enable CONFIG_HAVE_CXXINITIALIZE in cxxtest.

Comment (amogh-samsung) [reply build/configs/rtl8730e/cpp_loadable/defconfig:1271]: Updated, thankyou

File: build/configs/rtl8730e/loadable_apps_16M/defconfig [hunk 23]
```diff
@@ -463,11 +467,14 @@ CONFIG_GPIO=y
 # CONFIG_LCD is not set
 CONFIG_BCH=y
 CONFIG_RTC=y
+# CONFIG_RTC_DATETIME is not set
+# CONFIG_RTC_HIRES is not set
 # CONFIG_RTC_ALARM is not set
 CONFIG_RTC_DRIVER=y
-# CONFIG_RTC_HIRES is not set
 # CONFIG_RTC_IOCTL is not set
```
Comment (ewoodev) [inline build/configs/rtl8730e/loadable_apps_16M/defconfig:474]: Please enable CONFIG_RTC_IOCTL configuration

File: build/configs/rtl8730e/loadable_apps_16M/defconfig [hunk 24]
```diff
@@ -651,16 +645,16 @@ CONFIG_NET_ARP_TABLESIZE=10
...
 # CONFIG_NET_NETBUF_RECVINFO is not set
 CONFIG_NET_UDP_TTL=255
-CONFIG_NET_UDPLITE=y
+# CONFIG_NET_UDPLITE is not set
 CONFIG_NET_TCP=y
 CONFIG_NET_TCP_TTL=255
 CONFIG_NET_TCP_WND=26280
 # CONFIG_NET_WND_SCALE is not set
-CONFIG_NET_TCP_MAXRTX=12
+CONFIG_NET_TCP_MAXRTX=6
```
Comment (ewoodev) [inline build/configs/rtl8730e/loadable_apps_16M/defconfig:657]: Please change 
CONFIG_NET_TCP_MAXRTX=12
CONFIG_NET_TCP_SYNMAXRTX=6

to 

CONFIG_NET_TCP_MAXRTX=7
CONFIG_NET_TCP_SYNMAXRTX=7

File: build/configs/rtl8730e/loadable_apps_16M/defconfig [hunk 25]
```diff
@@ -1145,7 +1153,7 @@ CONFIG_LIBC_LONG_LONG=y
 CONFIG_LIBC_FLOATPRECISION=6
 CONFIG_LIBC_SCANSET=y
 # CONFIG_NOPRINTF_LONGLONG_TO_ASCII is not set
-# CONFIG_LIBC_IOCTL_VARIADIC is not set
+CONFIG_LIBC_IOCTL_VARIADIC=y
 # CONFIG_LIBC_WCHAR is not set
```
Comment (ewoodev) [inline build/configs/rtl8730e/loadable_apps_16M/defconfig:1157]: Please enable CONFIG_LIBC_WCHAR=y

**Review by ewoodev — COMMENTED** [review]
Please add below configuration
```
CONFIG_C99_BOOL8=y
CONFIG_HAVE_CXX=y
CONFIG_CXX_VERSION_11=y
# CONFIG_CXX_VERSION_14 is not set
# CONFIG_CXX_VERSION_17 is not set
CONFIG_CXX_NEWLONG=y

#
# LLVM C++ Library (libcxx)
#
CONFIG_LIBCXX=y
# CONFIG_LIBCXX_EXCEPTION is not set
CONFIG_LIBCXX_IOSTREAM_BUFSIZE=32
CONFIG_LIBCXX_HAVE_LIBSUPCXX=y
```

```
CONFIG_BINFMT_CONSTRUCTORS=y
```

**Comment by ewoodev** [comment]
I plan to deleate 8M defconfigs.
I'm sharing this because it may cause unnecessary work in your PR.
If you are comfortable with the task, you can clear the 8M defconfig first and then change it.

File: build/configs/rtl8730e/loadable_apps_16M/defconfig (hunk 23 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps_16M/defconfig:474]: Please let me know if you need these changes in loadable_apps only or all defconfigs?

File: build/configs/rtl8730e/loadable_apps_16M/defconfig (hunk 24 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps_16M/defconfig:657]: Please let me know if you need these changes in loadable_apps only or all defconfigs?

**Comment by amogh-samsung** [comment]
> Please add below configuration
> 
> ```
> CONFIG_C99_BOOL8=y
> CONFIG_HAVE_CXX=y
> CONFIG_CXX_VERSION_11=y
> # CONFIG_CXX_VERSION_14 is not set
> # CONFIG_CXX_VERSION_17 is not set
> CONFIG_CXX_NEWLONG=y
> 
> #
> # LLVM C++ Library (libcxx)
> #
> CONFIG_LIBCXX=y
> # CONFIG_LIBCXX_EXCEPTION is not set
> CONFIG_LIBCXX_IOSTREAM_BUFSIZE=32
> CONFIG_LIBCXX_HAVE_LIBSUPCXX=y
> ```
> 
> ```
> CONFIG_BINFMT_CONSTRUCTORS=y
> ```

you mean we need to enable C++ by default in loadable_apps defconfigs?

File: build/configs/rtl8730e/loadable_apps_16M/defconfig (hunk 24 above)
Comment (ewoodev) [reply build/configs/rtl8730e/loadable_apps_16M/defconfig]: we need to change all defconfig.

**Review by ewoodev — COMMENTED** [review]
Please update defconfig by auto-generate using menuconfig (Kconfig-Frontend)

and please disable CONFIG_GPIO

**Comment by amogh-samsung** [comment]
@ewoodev 
CONFIG_GPIO is disabled in all defconfigs
CONFIG_NET_TCP_MAXRTX and CONFIG_NET_TCP_SYNMAXRTX are set to 7 in all defconfigs

I have tried to align all defconfigs with menuconfig options, please let me know if you think anything is missing.

**Comment by ewoodev** [comment]
```
#
# Light Weight AI Framework
#
# CONFIG_LW_AIFW is not set

```

This is not a latest auto-generated by menuconfig
Please update again using menuconfig

**Review by ewoodev — APPROVED** [review]
Thank you for making a lot of changes
Looks good to me.

File: build/configs/rtl8730e/audio/defconfig [hunk 26]
```diff
@@ -71,7 +72,7 @@ CONFIG_ARM_HAVE_NEON=y
 # CONFIG_ARM_FPU_ABI_SOFT is not set
 CONFIG_ARM_DPFPU32=y
 CONFIG_ARM_NEON=y
-# CONFIG_ARM_HAVE_MPCORE is not set
+CONFIG_ARM_HAVE_MPCORE=y
```
Comment (ewoodev) [inline build/configs/rtl8730e/audio/defconfig:75]: oh please disable CONFIG_ARM_HAVE_MPCORE configuration,

please refer https://github.com/Samsung/TizenRT/commit/840fd41255b648b76412fb14871eace27205ede7

Comment (amogh-samsung) [reply build/configs/rtl8730e/audio/defconfig]: I disabled this in all defconfigs

File: build/configs/rtl8730e/audio/defconfig [hunk 27]
```diff
@@ -463,7 +464,7 @@ CONFIG_I2C_POLLED=y
 # CONFIG_I2C_TRACE is not set
 # CONFIG_I2C_WRITEREAD is not set
 # CONFIG_SPI is not set
-CONFIG_GPIO=y
+# CONFIG_GPIO is not set
```
Comment (Taejun-Kwon) [inline build/configs/rtl8730e/audio/defconfig:467]: it should be enabled to handle wake up

Comment (amogh-samsung) [reply build/configs/rtl8730e/audio/defconfig:467]: Audio defconfig diverged in between and I am not sure of which defconfig was used as reference for creating it.
So I am not sure about some other configs too in audio.
Please give me some more info and then I can modify audio defconfig as per latest requirements.

Comment (abhishek-samsung) [reply build/configs/rtl8730e/audio/defconfig:467]: except this config, rest all changes wont affect audio related code. So, please revert CONFIG_GPIO change.

**Review by ewoodev — APPROVED** [review]

**Review by ewoodev — COMMENTED** [review]
Please update base branch
It has conflicts

File: build/configs/rtl8730e/loadable_apps_16M/defconfig (hunk 25 above)
Comment (amogh-samsung) [reply build/configs/rtl8730e/loadable_apps_16M/defconfig:1174]: Done

**Review by ewoodev — APPROVED** [review]

**Review by sunghan-chang — APPROVED** [review]
